No podspec found for `path_provider_windows`

I have an ios app build through de azure pipelines, I have an error building the app, in the step flutter build ios, I have the next error <code>[!] No podspec found for `path_provider_windows` in `.symlinks/plugins/path_provider_windows/ios`</code>, looking for a reason, I find what that path doesn't have the podspect file in the plugin repository, but I don't know if that is right, and I don't know how to fix this issue 

## Steps to Reproduce
1. gem update fastlane
2. flutter pub get
3. sudo gem uninstall cocoapods -ax
4. sudo gem install cocoapods --pre
5. flutter build ios --build-name=$(version_name) --build-number=$(build_number)

## Logs

<details> 
<summary>logs</summary> 

```bash
2021-03-30T00:53:41.8195290Z ##[section]Starting: Build IOS App
2021-03-30T00:53:41.8206340Z ==============================================================================
2021-03-30T00:53:41.8206780Z Task         : Command line
2021-03-30T00:53:41.8207200Z Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
2021-03-30T00:53:41.8207570Z Version      : 2.182.0
2021-03-30T00:53:41.8207870Z Author       : Microsoft Corporation
2021-03-30T00:53:41.8208250Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
2021-03-30T00:53:41.8208750Z ==============================================================================
2021-03-30T00:53:42.0004290Z Generating script.
2021-03-30T00:53:42.0018720Z Script contents:
2021-03-30T00:53:42.0020790Z /Users/runner/hostedtoolcache/Flutter/2.0.3-stable/macos/flutter/bin/flutter build ios --build-name=5.0.4530 --build-number=20210330.03
2021-03-30T00:53:42.0022070Z ========================== Starting Command Output ===========================
2021-03-30T00:53:42.0055300Z [command]/bin/bash --noprofile --norc /Users/runner/work/_temp/0caf0513-ccf6-4f95-ac11-1ff84f7dd6d2.sh
2021-03-30T00:53:45.1695230Z Running "flutter pub get" in s...                                1,990ms
2021-03-30T00:53:49.3271000Z Building com.sivale.tarjetas for device (ios-release)...
2021-03-30T00:53:49.3494670Z Upgrading contents.xcworkspacedata
2021-03-30T00:53:49.8729730Z Automatically signing iOS for device deployment using specified development team in Xcode project: QAWSXS2H69
2021-03-30T00:53:51.8273460Z Running pod install...                                           1,052ms
2021-03-30T00:53:51.8290000Z CocoaPods' output:
2021-03-30T00:53:51.8291310Z â†³
2021-03-30T00:53:51.8294970Z       Preparing
2021-03-30T00:53:51.8295600Z 
2021-03-30T00:53:51.8296320Z     Analyzing dependencies
2021-03-30T00:53:51.8296930Z 
2021-03-30T00:53:51.8297710Z     Inspecting targets to integrate
2021-03-30T00:53:51.8299060Z       Using `ARCHS` setting to build architectures of target `Pods-Runner`: (``)
2021-03-30T00:53:51.8300800Z       Using `ARCHS` setting to build architectures of target `Pods-OneSignalNotificationServiceExtension`: (``)
2021-03-30T00:53:51.8301630Z 
2021-03-30T00:53:51.8302400Z     Finding Podfile changes
2021-03-30T00:53:51.8303170Z       A firebase_analytics
2021-03-30T00:53:51.8304010Z       A firebase_analytics_web
2021-03-30T00:53:51.8304530Z       A firebase_core
2021-03-30T00:53:51.8304950Z       A firebase_core_web
2021-03-30T00:53:51.8305340Z       A firebase_crashlytics
2021-03-30T00:53:51.8305680Z       A flutter_appauth
2021-03-30T00:53:51.8306700Z       A flutter_custom_tabs
2021-03-30T00:53:51.8307090Z       A flutter_secure_storage
2021-03-30T00:53:51.8307480Z       A flutter_webview_plugin
2021-03-30T00:53:51.8307800Z       A geocoder
2021-03-30T00:53:51.8308160Z       A integration_test
2021-03-30T00:53:51.8309140Z       A location
2021-03-30T00:53:51.8309860Z       A location_web
2021-03-30T00:53:51.8310280Z       A onesignal_flutter
2021-03-30T00:53:51.8310630Z       A package_info
2021-03-30T00:53:51.8310990Z       A path_provider
2021-03-30T00:53:51.8311360Z       A path_provider_linux
2021-03-30T00:53:51.8312270Z       A path_provider_macos
2021-03-30T00:53:51.8312630Z       A path_provider_windows
2021-03-30T00:53:51.8313030Z       A shared_preference_app_group
2021-03-30T00:53:51.8313420Z       A shared_preferences
2021-03-30T00:53:51.8313800Z       A shared_preferences_macos
2021-03-30T00:53:51.8314160Z       A shared_preferences_web
2021-03-30T00:53:51.8314540Z       A sqflite
2021-03-30T00:53:51.8314890Z       A url_launcher
2021-03-30T00:53:51.8315270Z       A url_launcher_macos
2021-03-30T00:53:51.8315630Z       A url_launcher_web
2021-03-30T00:53:51.8316450Z       - OneSignal
2021-03-30T00:53:51.8316670Z 
2021-03-30T00:53:51.8316990Z     Fetching external sources
2021-03-30T00:53:51.8317870Z     -> Fetching podspec for `firebase_analytics` from `.symlinks/plugins/firebase_analytics/ios`
2021-03-30T00:53:51.8318910Z     firebase_analytics: Using Firebase SDK version '7.3.0' defined in 'firebase_core'
2021-03-30T00:53:51.8321070Z     -> Fetching podspec for `firebase_analytics_web` from `.symlinks/plugins/firebase_analytics_web/ios`
2021-03-30T00:53:51.8322100Z     -> Fetching podspec for `firebase_core` from `.symlinks/plugins/firebase_core/ios`
2021-03-30T00:53:51.8323090Z     firebase_core: Using Firebase SDK version '7.3.0' defined in 'firebase_core'
2021-03-30T00:53:51.8324460Z     -> Fetching podspec for `firebase_core_web` from `.symlinks/plugins/firebase_core_web/ios`
2021-03-30T00:53:51.8325510Z     -> Fetching podspec for `firebase_crashlytics` from `.symlinks/plugins/firebase_crashlytics/ios`
2021-03-30T00:53:51.8326560Z     firebase_crashlytics: Using Firebase SDK version '7.3.0' defined in 'firebase_core'
2021-03-30T00:53:51.8327550Z     -> Fetching podspec for `flutter_appauth` from `.symlinks/plugins/flutter_appauth/ios`
2021-03-30T00:53:51.8328570Z     -> Fetching podspec for `flutter_custom_tabs` from `.symlinks/plugins/flutter_custom_tabs/ios`
2021-03-30T00:53:51.8329590Z     -> Fetching podspec for `flutter_secure_storage` from `.symlinks/plugins/flutter_secure_storage/ios`
2021-03-30T00:53:51.8330660Z     -> Fetching podspec for `flutter_webview_plugin` from `.symlinks/plugins/flutter_webview_plugin/ios`
2021-03-30T00:53:51.8331650Z     -> Fetching podspec for `geocoder` from `.symlinks/plugins/geocoder/ios`
2021-03-30T00:53:51.8332740Z     -> Fetching podspec for `integration_test` from `.symlinks/plugins/integration_test/ios`
2021-03-30T00:53:51.8333710Z     -> Fetching podspec for `location` from `.symlinks/plugins/location/ios`
2021-03-30T00:53:51.8334680Z     -> Fetching podspec for `location_web` from `.symlinks/plugins/location_web/ios`
2021-03-30T00:53:51.8335640Z     -> Fetching podspec for `onesignal_flutter` from `.symlinks/plugins/onesignal_flutter/ios`
2021-03-30T00:53:51.8336640Z     -> Fetching podspec for `package_info` from `.symlinks/plugins/package_info/ios`
2021-03-30T00:53:51.8337620Z     -> Fetching podspec for `path_provider` from `.symlinks/plugins/path_provider/ios`
2021-03-30T00:53:51.8339310Z     -> Fetching podspec for `path_provider_linux` from `.symlinks/plugins/path_provider_linux/ios`
2021-03-30T00:53:51.8340780Z     -> Fetching podspec for `path_provider_macos` from `.symlinks/plugins/path_provider_macos/ios`
2021-03-30T00:53:51.8342380Z     -> Fetching podspec for `path_provider_windows` from `.symlinks/plugins/path_provider_windows/ios`
2021-03-30T00:53:51.8343040Z     [!] No podspec found for `path_provider_windows` in `.symlinks/plugins/path_provider_windows/ios`
2021-03-30T00:53:51.8343730Z 
2021-03-30T00:53:51.8344670Z     /usr/local/lib/ruby/gems/2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/external_sources/path_source.rb:14:in `block in fetch'
2021-03-30T00:53:51.8345810Z     /usr/local/lib/ruby/gems/2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/user_interface.rb:64:in `section'
2021-03-30T00:53:51.8346930Z     /usr/local/lib/ruby/gems/2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/external_sources/path_source.rb:11:in `fetch'
2021-03-30T00:53:51.8348090Z     /usr/local/lib/ruby/gems/2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/installer/analyzer.rb:989:in `fetch_external_source'
2021-03-30T00:53:51.8349320Z     /usr/local/lib/ruby/gems/2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/installer/analyzer.rb:968:in `block (2 levels) in fetch_external_sources'
2021-03-30T00:53:51.8350470Z     /usr/local/lib/ruby/gems/2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/installer/analyzer.rb:967:in `each'
2021-03-30T00:53:51.8352100Z     /usr/local/lib/ruby/gems/2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/installer/analyzer.rb:967:in `block in fetch_external_sources'
2021-03-30T00:53:51.8353420Z     /usr/local/lib/ruby/gems/2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/user_interface.rb:64:in `section'
2021-03-30T00:53:51.8355370Z     /usr/local/lib/ruby/gems/2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/installer/analyzer.rb:966:in `fetch_external_sources'
2021-03-30T00:53:51.8357260Z     /usr/local/lib/ruby/gems/2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/installer/analyzer.rb:117:in `analyze'
2021-03-30T00:53:51.8359430Z     /usr/local/lib/ruby/gems/2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/installer.rb:414:in `analyze'
2021-03-30T00:53:51.8360930Z     /usr/local/lib/ruby/gems/2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/installer.rb:239:in `block in resolve_dependencies'
2021-03-30T00:53:51.8362630Z     /usr/local/lib/ruby/gems/2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/user_interface.rb:64:in `section'
2021-03-30T00:53:51.8363800Z     /usr/local/lib/ruby/gems/2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/installer.rb:238:in `resolve_dependencies'
2021-03-30T00:53:51.8364960Z     /usr/local/lib/ruby/gems/2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/installer.rb:160:in `install!'
2021-03-30T00:53:51.8366050Z     /usr/local/lib/ruby/gems/2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/command/install.rb:52:in `run'
2021-03-30T00:53:51.8367080Z     /usr/local/lib/ruby/gems/2.7.0/gems/claide-1.0.3/lib/claide/command.rb:334:in `run'
2021-03-30T00:53:51.8368120Z     /usr/local/lib/ruby/gems/2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/command.rb:52:in `run'
2021-03-30T00:53:51.8369140Z     /usr/local/lib/ruby/gems/2.7.0/gems/cocoapods-1.10.1/bin/pod:55:in `<top (required)>'
2021-03-30T00:53:51.8370010Z     /usr/local/lib/ruby/gems/2.7.0/bin/pod:23:in `load'
2021-03-30T00:53:51.8370920Z     /usr/local/lib/ruby/gems/2.7.0/bin/pod:23:in `<main>'
2021-03-30T00:53:51.8371150Z 
2021-03-30T00:53:51.8455010Z Error running pod install
2021-03-30T00:53:51.8848060Z ##[error]Bash exited with code '1'.
2021-03-30T00:53:51.8867450Z ##[section]Finishing: Build IOS App
```

</details>

@jmagman 
I updated the pod file, but the build failed with the following error
```
    1 warning generated.
    ld: warning: Could not find or use auto-linked framework 'Flutter'
    Undefined symbols for architecture armv7:
      "_OBJC_CLASS_$_FlutterStandardTypedData", referenced from:
          objc-class-ref in libsqflite.a(SqflitePlugin.o)
      "_OBJC_CLASS_$_FlutterError", referenced from:
          objc-class-ref in libfirebase_core.a(FLTFirebasePlugin.o)
          objc-class-ref in libflutter_appauth.a(FlutterAppauthPlugin.o)
          objc-class-ref in libgeocoder.a(GeocoderPlugin.o)
          objc-class-ref in liblocation.a(LocationPlugin.o)
          objc-class-ref in libonesignal_flutter.a(OSFlutterCategories.o)
          objc-class-ref in libpath_provider.a(FLTPathProviderPlugin.o)
          objc-class-ref in libshared_preference_app_group.a(SharedPreferenceAppGroupPlugin.o)
          ...
      "_OBJC_CLASS_$_FlutterEventChannel", referenced from:
          objc-class-ref in liblocation.a(LocationPlugin.o)
      "_OBJC_CLASS_$_FlutterViewController", referenced from:
          objc-class-ref in libintegration_test.a(IntegrationTestIosTest.o)
      "_FlutterMethodNotImplemented", referenced from:
          -[FLTFirebaseAnalyticsPlugin handleMethodCall:result:] in libfirebase_analytics.a(FLTFirebaseAnalyticsPlugin.o)
          -[FLTFirebaseCorePlugin handleMethodCall:result:] in libfirebase_core.a(FLTFirebaseCorePlugin.o)
          -[FLTFirebaseCrashlyticsPlugin handleMethodCall:result:] in libfirebase_crashlytics.a(FLTFirebaseCrashlyticsPlugin.o)
          -[FlutterAppauthPlugin handleMethodCall:result:] in libflutter_appauth.a(FlutterAppauthPlugin.o)
          -[GDBCustomTabsPlugin handleMethodCall:result:] in libflutter_custom_tabs.a(CustomTabsPlugin.o)
          -[FlutterSecureStoragePlugin handleMethodCall:result:] in libflutter_secure_storage.a(FlutterSecureStoragePlugin.o)
          -[FlutterWebviewPlugin handleMethodCall:result:] in libflutter_webview_plugin.a(FlutterWebviewPlugin.o)
          ...
      "_OBJC_CLASS_$_FlutterMethodChannel", referenced from:
          objc-class-ref in libfirebase_analytics.a(FLTFirebaseAnalyticsPlugin.o)
          objc-class-ref in libfirebase_core.a(FLTFirebaseCorePlugin.o)
          objc-class-ref in libfirebase_crashlytics.a(FLTFirebaseCrashlyticsPlugin.o)
          objc-class-ref in libflutter_appauth.a(FlutterAppauthPlugin.o)
          objc-class-ref in libflutter_custom_tabs.a(CustomTabsPlugin.o)
          objc-class-ref in libflutter_secure_storage.a(FlutterSecureStoragePlugin.o)
          objc-class-ref in libflutter_webview_plugin.a(FlutterWebviewPlugin.o)
          ...
    ld: symbol(s) not found for architecture armv7
   clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/packages/flutter_tools/lib/src/macos/cocoapods.dart
+++ b/packages/flutter_tools/lib/src/macos/cocoapods.dart
@@ -162,6 +162,7 @@ class CocoaPods {
     if (!xcodeProject.podfile.existsSync()) {
       throwToolExit('Podfile missing');
     }
+    _warnIfPodfileOutOfDate(xcodeProject);
     bool podsProcessed = false;
     if (_shouldRunPodInstall(xcodeProject, dependenciesChanged)) {
       if (!await _checkPodCondition()) {
@@ -170,7 +171,6 @@ class CocoaPods {
       await _runPodInstall(xcodeProject, buildMode);
       podsProcessed = true;
     }
-    _warnIfPodfileOutOfDate(xcodeProject);
     return podsProcessed;
   }
 

--- a/packages/flutter_tools/test/general.shard/macos/cocoapods_test.dart
+++ b/packages/flutter_tools/test/general.shard/macos/cocoapods_test.dart
@@ -20,6 +20,7 @@ import 'package:test/fake.dart';
 
 import '../../src/common.dart';
 import '../../src/context.dart';
+import '../../src/fake_process_manager.dart';
 
 typedef InvokeProcess = Future<ProcessResult> Function();
 
@@ -322,6 +323,7 @@ void main() {
         buildMode: BuildMode.debug,
       ), throwsToolExit(message: 'CocoaPods not installed or not in valid state'));
       expect(fakeProcessManager.hasRemainingExpectations, isFalse);
+      expect(fakeProcessManager, hasNoRemainingExpectations);
     });
 
     testWithoutContext('throwsToolExit if CocoaPods install is broken', () async {
@@ -332,17 +334,10 @@ void main() {
         buildMode: BuildMode.debug,
       ), throwsToolExit(message: 'CocoaPods not installed or not in valid state'));
       expect(fakeProcessManager.hasRemainingExpectations, isFalse);
+      expect(fakeProcessManager, hasNoRemainingExpectations);
     });
 
     testWithoutContext('exits if Podfile creates the Flutter engine symlink', () async {
-      pretendPodIsInstalled();
-      pretendPodVersionIs('1.10.0');
-      fakeProcessManager.addCommand(
-        const FakeCommand(
-          command: <String>['pod', 'install', '--verbose'],
-        ),
-      );
-
       fileSystem.file(fileSystem.path.join('project', 'ios', 'Podfile'))
         ..createSync()
         ..writeAsStringSync('Existing Podfile');
@@ -355,17 +350,10 @@ void main() {
         xcodeProject: projectUnderTest.ios,
         buildMode: BuildMode.debug,
       ), throwsToolExit(message: 'Podfile is out of date'));
+      expect(fakeProcessManager, hasNoRemainingExpectations);
     });
 
     testWithoutContext('exits if iOS Podfile parses .flutter-plugins', () async {
-      pretendPodIsInstalled();
-      pretendPodVersionIs('1.10.0');
-      fakeProcessManager.addCommand(
-        const FakeCommand(
-          command: <String>['pod', 'install', '--verbose'],
-        ),
-      );
-
       fileSystem.file(fileSystem.path.join('project', 'ios', 'Podfile'))
         ..createSync()
         ..writeAsStringSync('plugin_pods = parse_KV_file(\'../.flutter-plugins\')');
@@ -374,6 +362,7 @@ void main() {
         xcodeProject: projectUnderTest.ios,
         buildMode: BuildMode.debug,
       ), throwsToolExit(message: 'Podfile is out of date'));
+      expect(fakeProcessManager, hasNoRemainingExpectations);
     });
 
     testWithoutContext('prints warning if macOS Podfile parses .flutter-plugins', () async {
@@ -396,6 +385,7 @@ void main() {
 
       expect(logger.errorText, contains('Warning: Podfile is out of date'));
       expect(logger.errorText, contains('rm macos/Podfile'));
+      expect(fakeProcessManager, hasNoRemainingExpectations);
     });
 
     testWithoutContext('throws, if Podfile is missing.', () async {

